### PR TITLE
Fix tests failing in the browser and other test improvements.

### PR DESCRIPTION
**Pull request by jansiegel**
### Context
Some tests were only failing in the browser, this change should fix them.

**Note: when running tests in the browser, it's important not to do anything and keep the tab focused, as, for example, changing the focus to another window may mess up the `focus` event recognition, and simply moving the cursor might close the comments pop-up, resulting in a failed test.**

Additionally, I modified the `ContextMenu` plugin and `DateEditor` to prevent leaving behind containers after destroying the Handsontable instances.

### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)

### CODE REVIEW

File: src/3rdparty/walkontable/test/spec/settings/stretchH.spec.js [hunk 1]
```diff
@@ -69,12 +69,15 @@ describe('stretchH option', () => {
...
     window.dispatchEvent(evt);
+    wt.draw();
 
     await sleep(300);
 
     const currentTableWidth = $table.outerWidth();
     expect(currentTableWidth).toBeAroundValue($table[0].clientWidth);
     expect(currentTableWidth).toBeGreaterThan(initialTableWidth);
+
+    done();
```
Comment (budnix) [inline src/3rdparty/walkontable/test/spec/settings/stretchH.spec.js]: `done()` call is not necessary when `async/await` is used.

File: src/editors/dateEditor.js [hunk 2]
```diff
@@ -74,6 +74,7 @@ class DateEditor extends TextEditor {
    */
   destroyElements() {
     this.$datePicker.destroy();
+    this.datePicker.remove();
```
Comment (budnix) [inline src/editors/dateEditor.js]: `.remove()` method is not supported by IE (https://developer.mozilla.org/en-US/docs/Web/API/ChildNode/remove)

File: src/plugins/contextMenu/menu.js [hunk 3]
```diff
@@ -265,6 +265,7 @@ class Menu {
     this.close();
     this.parentMenu = null;
     this.eventManager.destroy();
+    this.container.remove();
```
Comment (budnix) [inline src/plugins/contextMenu/menu.js]: `.remove()` method is not supported by IE (https://developer.mozilla.org/en-US/docs/Web/API/ChildNode/remove). Additionally, can I ask you to add/update test which will test if that element was removed?

File: src/3rdparty/walkontable/test/spec/settings/stretchH.spec.js (hunk 1 above)
Comment (jansiegel) [reply src/3rdparty/walkontable/test/spec/settings/stretchH.spec.js]: Without `done` the test would timeout before the async part was finished:
image

After updating Jasmine to the latest version this seem to not happen anymore, so I removed it.

**Comment by jansiegel** [comment]
Updates after the CR:
- Updated Jasmine to the latest version (`2.5.2` -> `^3.4.0`)
- More leftover DOM elements removed
- Fixed Firefox and retina problems
- Temporarily disabled some tests - to be re-enabled after fixing  #6083 and #6085 (and possible remove the ones related with table captions)

If I'm not mistaken, these changes seem to have fixed (or at least worked around) the Chrome memory leak problem.

File: src/validators/dateValidator.js [hunk 4]
```diff
@@ -18,9 +18,9 @@ export default function dateValidator(value, callback) {
   if (valueToValidate === null || valueToValidate === void 0) {
     valueToValidate = '';
   }
-  let isValidDate = moment(new Date(valueToValidate)).isValid() || moment(valueToValidate, this.dateFormat || dateEditor.defaultDateFormat, true).isValid();
-  // is it in the specified format
+
   let isValidFormat = moment(valueToValidate, this.dateFormat || dateEditor.defaultDateFormat, true).isValid();
+  let isValidDate = moment(new Date(valueToValidate)).isValid() || isValidFormat;
```
Comment (budnix) [inline src/validators/dateValidator.js]: A good catch 👍

**Pull request merged**